test/Core.Test/AutoFixture/OrganizationFixtures.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using AutoFixture;
+using Bit.Core.Enums;
+using Bit.Core.Models.Business;
+using Bit.Core.Models.Data;
+using Bit.Core.Models.Table;
+using Bit.Core.Test.AutoFixture.Attributes;
+using Bit.Core.Utilities;
+
+namespace Bit.Core.Test.AutoFixture.OrganizationFixtures
+{
+    internal class PaidOrganization : ICustomization
+    {
+        public PlanType CheckedPlanType { get; set; }
+        public void Customize(IFixture fixture)
+        {
+            var validUpgradePlans = StaticStore.Plans.Where(p => p.Type != Enums.PlanType.Free && !p.Disabled).Select(p => p.Type).ToList();
+            var lowestActivePaidPlan = validUpgradePlans.First();
+            CheckedPlanType = CheckedPlanType.Equals(Enums.PlanType.Free) ? lowestActivePaidPlan : CheckedPlanType;
+            validUpgradePlans.Remove(lowestActivePaidPlan);
+            fixture.Customize<Organization>(composer => composer
+                .With(o => o.PlanType, CheckedPlanType));
+            fixture.Customize<OrganizationUpgrade>(composer => composer
+                .With(ou => ou.Plan, validUpgradePlans.First()));
+        }
+    }
+
+    internal class FreeOrganizationUpgrade : ICustomization
+    {
+        public void Customize(IFixture fixture)
+        {
+            fixture.Customize<Organization>(composer => composer
+                .With(o => o.PlanType, PlanType.Free));
+
+            var plansToIgnore = new List<PlanType> { PlanType.Free, PlanType.Custom };
+            var validPlans = StaticStore.Plans.Where(p => !plansToIgnore.Contains(p.Type) && !p.Disabled).Select(p => p.Type).ToList();
+            fixture.Customize<OrganizationUpgrade>(composer => composer
+                .With(ou => ou.Plan, validPlans.Last()));
+            fixture.Customize<Organization>(composer => composer
+                .Without(o => o.GatewaySubscriptionId));
+        }
+    }
+    internal class OrganizationInvite : ICustomization
+    {
+        public OrganizationUserType InviteeUserType { get; set; }
```

Comment:
Is the intended behavior to set these to null by default? If not, please set these properties as mandatory params in a constructor.

Reply:
The intended behavior is to set these to `OranizationUserType.Owner` (via its int value `0`) by default. The attribute constructor specifies the default value for this in their init. The thought behind making these optional parameters with default values was to say "it doesn't matter what these values are" while also making sure AutoFixture isn't setting illogical or impossible combinations.

Reply:
I didn't realize the types were enums and thought they were classes. My mistake. I think the same question applies to blob, but that one makes more sense to default to null.